### PR TITLE
Fix `WindowInfo.containerSize` without `platformLayers` flag

### PR DESCRIPTION
**Pull request by dima-avdeev-jb**
This PR fixes Dialog position when platformLayers=false.
We applied same logic as for Desktop sourceSet with `PlatformWindowContext`

coauthored with @MatkovIvan

### CODE REVIEW

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    private val _windowInfo = WindowInfoImpl()
+    val windowInfo: WindowInfo get() = _windowInfo
+
+    private var _windowContainer: UIView? = null
+
+    /**
+     * Indicates whether the window is transparent or not.
+     * Used for determining the right blending mode for [Dialog]'s scrim.
+     */
+    var isWindowTransparent: Boolean by mutableStateOf(false)
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: Unused, please remove

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+        }
+    }
+
+    /**
+     * Calculates the offset of the given [container] within the window.
+     * It uses [_windowContainer] as a reference for window coordinate space.
+     *
+     * @param container The container component whose offset needs to be calculated.
+     * @return The offset of the container within the window as an [IntOffset] object.
+     */
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]:
```suggestion
    /**
     * Calculates the bounds of the given [container] within the window.
     * It uses [_windowContainer] as a reference for window coordinate space.
     *
     * @param container The container component whose bounds need to be calculated.
     * @return The bounds of the container within the window as an [IntRect] object.
     */
```

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    fun boundsInWindow(container: UIView): IntRect {
+        fun getSystemDensity(): Density {
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: Can we share this code as internal `UIView` extension?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+            )
+        } else {
+            container.bounds
+        }.useContents {
+            with(density) {
+                toDpRect().toRect().roundToIntRect()
+            }
+        }
+    }
+
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]:
```suggestion
```

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt [hunk 5]
```diff
@@ -99,6 +99,9 @@ internal class ComposeContainer(
     private val layers: MutableList<UIViewComposeSceneLayer> = mutableListOf()
     private val layoutDirection get() = getLayoutDirection()
 
+    @OptIn(ExperimentalComposeApi::class)
+    private val windowContainer: UIView get() = if (configuration.platformLayers) view.window!! else view
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt]:
```suggestion
    private val windowContainer: UIView
        get() = if (configuration.platformLayers) checkNotNull(view.window) {
            "ComposeUIViewController.view should be attached to window"
        } else view
```

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt [hunk 6]
```diff
@@ -172,13 +175,14 @@ internal class ComposeContainer(
             "ComposeUIViewController.view should be attached to window"
         }
         val scale = window.screen.scale
```
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt]: Shouldn't we drop accessing the winding here?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt (hunk 4 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: GitHub offers to add empty lines at the end of file.
It helps to properly view new lines of code at the end of file.

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt (hunk 1 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: Done

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt (hunk 4 above)
Comment (MatkovIvan) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: It's not EOF

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt (hunk 3 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: Done

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt (hunk 4 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/platform/PlatformWindowContext.uikit.kt]: Done

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt (hunk 6 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt]: What do you mean?
Should I remove window usage from here?

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/Helpers.kt
Comment (MatkovIvan) [inline compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/Helpers.kt:1]: `ui/Helpers.kt` -> `ui/uikit/Extensions.uikit.kt`

Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/Helpers.kt:1]: Done

File: compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt (hunk 6 above)
Comment (dima-avdeev-jb) [reply compose/ui/ui/src/uikitMain/kotlin/androidx/compose/ui/window/ComposeContainer.uikit.kt]: Done

**Review by MatkovIvan — APPROVED** [review]

**Pull request merged**